doc/api/stream.md

```diff
@@ -1211,6 +1229,24 @@ added: v12.3.0
 
 Getter for the property `objectMode` of a given `Readable` stream.
 
+##### `readable.updateReadableHighWaterMark(highwaterMark)`
+<!-- YAML
+added: REPLACEME
+-->
+
+* `highwaterMark` {number} new highwaterMark
+
+Update the value of `highWaterMark` of a given `Readable` stream.
+
+##### `readable.updateReadableObjectMode(objectMode)`
```

Comment:
As explained in the comment I am also -1 on this. I think that even if we find a compelling use case for switching a stream's mode interim it would still be very confusing and foot-gunny.